File: leetcode/python/746.py
class Solution:
    def minCostClimbingStairs(self, cost: List[int]) -> int:

        for i, _ in enumerate(cost):
            if i < 2:
                continue

            if cost[i - 1] > cost[i - 2]:
                cost[i] += cost[i - 2]
            else:
                cost[i] += cost[i - 1]

        return cost[-1] if cost[-1] < cost[-2] else cost[-2]

File: leetcode/go/746.go
func minCostClimbingStairs(cost []int) int {

	temp := make([]int, len(cost))
	temp[0] = cost[0]
	temp[1] = cost[1]

	for i := 2; i < len(cost); i++ {
		if temp[i-1] > temp[i-2] {
			temp[i] = temp[i-2] + cost[i]
		} else {
			temp[i] = temp[i-1] + cost[i]
		}
	}

	if temp[len(cost)-1] > temp[len(cost)-2] {
		return temp[len(cost)-2]
	} else {
		return temp[len(cost)-1]
	}
}
